.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,22 @@
+##### Motivation for the changes
+
+<!-- An issue reference or explanation why the change is desirable. -->
+<!-- This speeds up step 1 of the process. -->
+
+
+
+
+
+
+##### Checklist for maintainers
```

Comment:
I think we should be more precise about what we want to have here.

```suggestion
# Motivation
<!-- Briefly explain what the change is about and why it is desirable. -->

# Context
<!-- Provide context. Reference open issues if available. -->

<!-- Non-trivial change: Briefly outline the implementation strategy. -->

<!-- Invasive change: Discuss alternative designs or approaches you considered. -->

<!-- Large change: Provide instructions to reviewers how to read the diff. -->

# Checklist for maintainers
```

While it's probably undesirable to have large headings in the description, it also doesn't really hurt to make the sections more distinct. After all, it's just a presentational aspect.

Reply:
It may even be worthy to dedicate a whole (potentially blankable) section explicitly to design. It nudges people to possibly at least quickly revise their drawing board when they started to hack out  of an itch or momentum and feel really enthusiastic about their adrenaline rush.

```
# Design

<!-- Explain the implementation route and design that you've chosen and why. -->
```

Reply:
That was my first idea, too, but that requires additional work for trivial PRs, which I think we'd want to have many of and which should require a trivial amount of effort. If the PR is complex anyway, authors may as well be inclined to make a proper write-up.